core/src/main/scala/cats/MonadReader.scala

```diff
@@ -0,0 +1,14 @@
+package cats
+
+/** A monad that has the ability to read from an environment. */
+trait MonadReader[F[_, _], R] extends Monad[F[R, ?]] {
+  /** Get the environment */
+  def ask: F[R, R]
+
+  /** Modify the environment */
+  def local[A](f: R => R)(fa: F[R, A]): F[R, A]
```

Comment:
Fair enough, I often find myself really liking the function standalone in the last parameter list for the exact reason you describe above.

I'll wait for others to chime in with their thoughts, if folks like it as the 2nd/last parameter I'll happily make the change :-)